operate/webapp/src/main/java/io/camunda/operate/webapp/zeebe/operation/adapter/ServicesBasedAdapter.java

```diff
@@ -0,0 +1,380 @@
+import io.camunda.service.ProcessInstanceServices.ProcessInstanceModifyRequest;
+import io.camunda.service.ResourceServices;
+import io.camunda.service.ResourceServices.ResourceDeletionRequest;
+import io.camunda.service.exception.CamundaBrokerException;
+import io.camunda.webapps.schema.entities.flownode.FlowNodeState;
+import io.camunda.zeebe.broker.client.api.BrokerErrorException;
+import io.camunda.zeebe.broker.client.api.BrokerRejectionException;
+import io.camunda.zeebe.gateway.rest.RequestMapper;
+import io.camunda.zeebe.protocol.impl.encoding.MsgPackConverter;
+import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceMigrationMappingInstruction;
+import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceModificationActivateInstruction;
+import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceModificationTerminateInstruction;
+import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceModificationVariableInstruction;
+import io.camunda.zeebe.protocol.record.ErrorCode;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.function.Function;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+@Component
+@ConditionalOnOperateCompatibility(enabled = "false", matchIfMissing = true)
+public class ServicesBasedAdapter implements OperateServicesAdapter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ServicesBasedAdapter.class);
+
+  private final ProcessInstanceServices processInstanceServices;
+  private final ResourceServices resourceServices;
+  private final ElementInstanceServices elementInstanceServices;
+  private final JobServices<?> jobServices;
+  private final IncidentServices incidentServices;
+  private final FlowNodeInstanceReader flowNodeInstanceReader;
+
+  public ServicesBasedAdapter(
+      final ProcessInstanceServices processInstanceServices,
+      final ResourceServices resourceServices,
+      final ElementInstanceServices elementInstanceServices,
+      final JobServices<?> jobServices,
+      final IncidentServices incidentServices,
+      final FlowNodeInstanceReader flowNodeInstanceReader) {
+    this.processInstanceServices = processInstanceServices;
+    this.resourceServices = resourceServices;
+    this.elementInstanceServices = elementInstanceServices;
+    this.jobServices = jobServices;
+    this.incidentServices = incidentServices;
+    this.flowNodeInstanceReader = flowNodeInstanceReader;
+  }
+
+  @Override
+  public void deleteResource(final long resourceKey, final String operationId) {
+    executeCamundaServiceAuthenticated(
```

Comment:
@koevskinikola, is there a reason why it must authenticate the broker request? If I am not mistaken, these (batch/single) operations are always executed asynchronously. In that case, there is no authenticated user anymore. (The authenticated user is not preserved)

Reply:
Hey @romansmirnov, no specific reason. I've been guiding myself from the Tasklist PR you raised earlier (#25799). It wasn't that clear how we differentiate between authenticated and anonymous execution there, so I thought it would be safer to execute the requests authenticated.

Is my understanding correct that since the user was already authenticated in Operate, any requests sent to Zeebe can already be executed anonymously?

Reply:
I refactored the code. The operations are now executed with an anonymous user.